File: app/src/main/java/fr/clerc/tp3/RecyclerViewAdapter.java
package fr.clerc.tp3;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

import fr.clerc.tp3.model.EmployeeData;


public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder> {
    private final ArrayList<EmployeeData> employeeList;
    private final OnItemClickListener listener;

    public interface OnItemClickListener {
        void onItemClick(View view, EmployeeData employee);
    }

    public RecyclerViewAdapter(ArrayList<EmployeeData> employeeList, OnItemClickListener listener) {
        this.employeeList = employeeList;
        this.listener = listener;
    }

    @Override
    public int getItemCount() {
        return this.employeeList.size();
    }

    public void addEmployeeList(List<EmployeeData> employeeList) {
        this.employeeList.addAll(employeeList);
        notifyDataSetChanged();
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_employee, parent, false);
        return new RecyclerViewAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, final int position) {
        final EmployeeData employeeData = employeeList.get(position);

        holder.name.setText(employeeData.getEmployeeName());
        holder.age.setText(String.valueOf(employeeData.getEmployeeAge()));
        holder.salary.setText(String.valueOf(employeeData.getEmployeeSalary()));

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                listener.onItemClick(v, employeeData);
            }
        });
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        final View itemView;
        final TextView name;
        final TextView age;
        final TextView salary;

        ViewHolder(View view) {
            super(view);
            itemView = view;
            name = view.findViewById(R.id.item_name);
            age = view.findViewById(R.id.item_age);
            salary = view.findViewById(R.id.item_salary);
        }
    }
}

File: app/src/main/java/fr/clerc/tp3/kotlin/RecyclerViewAdapter.kt
package fr.clerc.tp3.kotlin

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import fr.clerc.tp3.R
import fr.clerc.tp3.kotlin.model.EmployeeData
import kotlinx.android.synthetic.main.item_employee.view.*

class RecyclerViewAdapter(
    private val itemList: ArrayList<EmployeeData>,
    val listener: (EmployeeData) -> Unit
) : RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder>() {

    override fun getItemCount(): Int {
        return itemList.size
    }

    fun addEmployee(list: List<EmployeeData>) {
        itemList.clear()
        itemList.addAll(list)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_employee, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val employee = itemList[position]

        holder.name.text = employee.employeeName
        holder.age.text = employee.employeeAge.toString()
        holder.salary.text = employee.employeeSalary.toString()

        holder.itemView.setOnClickListener {
            listener(employee)
        }
    }

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val name: TextView = view.item_name
        val age: TextView = view.item_age
        val salary: TextView = view.item_salary
    }
}